internal/shared/util/http/certpoolwatcher.go

```diff
@@ -33,77 +36,114 @@ func (cpw *CertPoolWatcher) Get() (*x509.CertPool, int, error) {
 	return cpw.pool.Clone(), cpw.generation, nil
 }
 
-func (cpw *CertPoolWatcher) Done() {
-	cpw.done <- true
+// Change the restart behavior
+// The default is to os.Exit(0) when a change to SSL_CERT_DIR or SSL_CERT_FILE
```

Comment:
I think I'd prefer if the restart func would be a pointer, and default to nil.  That way, it would be much more explicit when servicing its use-case, since the change of behavior to do os.Exit(0) would be associated with that area of the code. 

At this point I'm aware of one use-case serviced by this functionality, so it wouldn't lead to code duplication.  This is evident in the cpw.checkForRestart function, where we only check for conditions associated with the intially-unavailable-so-permanently-un-discoverable CA cert.

Reply:
I believe a `func` is considered a pointer, and defaults to `nil`, so need to change the type, just need to change where it is assigned.
But I did put the default to effectively be `os.Exit(0)`, so that I don't have to check the pointer, but it's only one place.

So, you are suggesting the assignment occur where the `CertPoolWatcher` is created?